core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -571,22 +598,68 @@ public CacheStatusResponse doJoin(Address joiner, CacheJoinInfo joinInfo) throws
                   "members = %s, joiners = %s", joiner, cacheName, expectedMembers, joiners);
             return new CacheStatusResponse(null, currentTopology, stableTopology, availabilityMode);
          }
-         if (isFirstMember) {
-            // This node was the first to join. We need to install the initial CH
-            CacheTopology initialTopology = createInitialCacheTopology();
+         if (status == ComponentStatus.INSTANTIATED) {
+            if (persistentState.isPresent()) {
+               if (trace) log.tracef("Node %s joining. Attempting to reform previous cluster", joiner);
+               // We can only allow this to proceed if we have a complete cluster
+               CacheTopology topology = restoreCacheTopology(persistentState.get(), joinInfo.getPersistentStateChecksum());
+               if (topology != null) {
+                  // Change our status
+                  status = ComponentStatus.RUNNING;
+                  clusterTopologyManager.broadcastTopologyUpdate(cacheName, topology, availabilityMode, isTotalOrder(), isDistributed());
+                  clusterTopologyManager.broadcastStableTopologyUpdate(cacheName, topology, isTotalOrder(), isDistributed());
```

Comment:
Actually I put the comment on the wrong line, I was looking at the `else` branch, and at the time I was assuming that you're sending the response with the initial topology and with `DEGRADED_MODE` to the first node that joined. 

Now that I've looked closer into `restoreCacheTopology`, I think you do need both broadcasts.